Gems/Atom/RPI/Code/Source/RPI.Edit/Material/MaterialTypeSourceData.cpp

```diff
@@ -290,6 +309,20 @@ namespace AZ
             materialTypeAssetCreator.SetElevateWarnings(elevateWarnings);
             materialTypeAssetCreator.Begin(assetId);
 
+            // Set materialtype version and add each version update object into MaterialTypeAsset.
+            materialTypeAssetCreator.SetVersion(m_version);
+            for (const auto& versionUpdate : m_versionUpdates)
+            {
+                MaterialVersionUpdate materialVersionUpdate;
+                for (const auto& action : versionUpdate.m_actions)
+                {
+                    materialVersionUpdate.AddAction(MaterialVersionUpdate::Action(AZ::Name{ "rename" }, {
+                        { AZ::Name{ "from" }, AZ::Name{ action.m_renameFrom } },
+                        { AZ::Name{ "to" }, AZ::Name{ action.m_renameTo } } }));
+                }
```

Comment:
Constructing and saving the name variables outside of the loop would be more efficient so that you don't have to reconstruct them every iteration

Reply:
Done!